ferrocene/tools/self-test/src/binaries.rs

```diff
@@ -25,15 +25,18 @@ fn check_binary(
     name: &str,
     hash: CommitHashOf,
 ) -> Result<(), Error> {
+    /// File permission the binary should have. The symbolic value is `r-xr-xr-x`.`
+    const MODE: u32 = 0o555;
+
     let bin_dir = sysroot.join("bin");
     let bin = bin_dir.join(name);
 
+    // check that `bin` is a file and has the correct permissions
     match std::fs::metadata(&bin) {
         Ok(metadata) => {
             if !metadata.is_file() {
                 return Err(Error::MissingBinary { directory: bin_dir, name: name.into() });
-            }
-            if metadata.permissions().mode() & 0o555 != 0o555 {
+            } else if metadata.permissions().mode() & MODE != MODE {
```

Comment:
I would not else here, since the if block above returns from the function

Reply:
Why? Both blocks do return from the function. It can only ever be one of them, never both.

Reply:
the `else` is redundant because it does not change the logic, since a `return` means the execution cannot continue anyways

Reply:
I know. But the else makes it more compact imo. And also it enables us to remove the `return` because we also move it into a function (in a later commit).

Reply:
me fan of removing the `return`, so 👍